### PR TITLE
Basic field edit form (add/remove/sort lists)

### LINKED ISSUE #3677: Infinite Corridor: Field basic info edit form
Create an edit form for users to select a field's featured_list and choose/order the field's other user lists.

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #3677

#### What's this PR do?
Adds a form for specifying the lists and featured list for a field channel.

#### How should this be manually tested?
- Adjust the following in your .env file:
   ```
   OPEN_DISCUSSIONS_HOSTNAME=od.odl.local
   ENABLE_INFINITE_CORRIDOR=True
   ```
- Go to `../learn/`, and use the bookmark icons on items to create new public lists and add courses to each of them (make sure you choose "Public" for privacy level when creating new lists).
- Create some fields if you don't have any:
  - Go to `/api/v0/fields/` and post something like this JSON to create one:
    ```
    {
       "name": "physics",
       "title": "Physics"
    }
    ```  
- Go to `../infinite/fields/<field_name>/manage/`
- You should see a form with "Featured List" and "Lists" inputs.  If your field already has values for these, they should be visible here.
- Use the inputs to change/add/remove/reorder lists.  Start typing in each text input field and check that the available options gets narrowed down to those that match your text.
- Submit the form, you should end up back to the field's display page and your changes should be reflected there.

#### Any background context you want to provide?
The unit testing coverage is not as good as it could be, in particular I skipped testing the reordering of components via dragging because I couldn't seem to emulate that with RTL.

#### Screenshots (if appropriate)
Screen Shot 2022-08-10 at 7 35 38 AM

### CODE REVIEW

File: open_discussions/settings.py [hunk 1]
```diff
@@ -556,7 +556,7 @@
     DEFAULT_FILE_STORAGE = "storages.backends.s3boto.S3BotoStorage"
 
 IMAGEKIT_SPEC_CACHEFILE_NAMER = "imagekit.cachefiles.namers.source_name_dot_hash"
-IMAGEKIT_CACHEFILE_DIR = get_string("IMAGEKIT_CACHEFILE_DIR", "")
+# IMAGEKIT_CACHEFILE_DIR = get_string("IMAGEKIT_CACHEFILE_DIR", "")
```
Comment (gumaerc) [inline open_discussions/settings.py:559]: Why is this commented out?

**Review by gumaerc — APPROVED** [review]
Everything worked properly for me. I tested setting fields in all the spots, reordering, etc.  Just one question about a commented setting and wondering if that's intentional:

File: open_discussions/settings.py (hunk 1 above)
Comment (mbertrand) [reply open_discussions/settings.py:559]: Shouldn't be there, it's dupe of the line above, I'll remove it.

Comment (mbertrand) [reply open_discussions/settings.py]: Mistake, uncommented it

**Comment by mbertrand** [comment]
These suggested changes just came in, will work on that next:

**Comment by mbertrand** [comment]
Updated screenshot:
Screen Shot 2022-08-12 at 1 00 47 PM

**Review by gumaerc — APPROVED** [review]
:+1: LGTM!

**Pull request merged**